virttest/utils_test/libvirt.py

```diff
@@ -2589,29 +2591,33 @@ def update_vm_disk_source(vm_name, disk_source_path, source_type="file"):
     if not os.path.isdir(disk_source_path):
         logging.error("Require disk source path!!")
         return False
+
     # Prepare to update VM first disk source file
     vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
     devices = vmxml.devices
     disk_index = devices.index(devices.by_device_tag('disk')[0])
     disks = devices[disk_index]
-    disk_source = disks.source.get_attrs().get(source_type)
-    logging.debug("The disk source file of the VM: %s", disk_source)
-    if not os.path.exists(disk_source):
-        logging.error("The disk source doesn't exist!!")
-        return False
+    # Generate a disk image name if it doesn't exist
+    if not disk_image_name:
+        disk_source = disks.source.get_attrs().get(source_type)
+        logging.debug("The disk source file of the VM: %s", disk_source)
```

Comment:
@PandaWei 
Yes, check the disk exist is useful. But as the author is taking a long time leave, I think we can merge this patch first then add the check part if you want. What do you think?